conformance/ConformanceApi.fsd

```diff
@@ -130,6 +130,111 @@ service ConformanceApi
 		name: string;
 	}
 
+	/// Gets external widgets.
+	[http(method: GET, path: "/externalWidgets")]
+	[tag(name: externalWidgets)]
+	method getExternalWidgets
+	{
+		/// The query.
+		[http(name: "q")]
+		query: string;
+	}:
+	{
+		/// The external widgets.
+		externalWidgets: ExternalWidget[];
+	}
+
+	/// Creates a new external widget.
+	[http(method: POST, path: "/externalWidgets")]
+	[tag(name: externalWidgets)]
+	method createExternalWidget
+	{
+		/// The external widget to create.
+		[http(from: body)]
+		externalWidget: ExternalWidget;
+	}:
+	{
+		/// The created external widget.
+		[http(from: body, code: 201)]
+		externalWidget: ExternalWidget;
+
+		/// The URL of the created external widget.
+		[http(from: header, name: Location)]
+		url: string;
+
+		/// The ETag of the created external widget.
+		[http(from: header)]
+		eTag: string;
+	}
+
+	/// Gets the specified external widget.
+	[http(method: GET, path: "/externalWidgets/{id}")]
+	[tag(name: externalWidgets)]
+	method getExternalWidget
+	{
+		/// The external widget ID.
+		[required, validate(value: 0..)]
+		id: int32;
+
+		/// Don't get the external widget if it has this ETag.
+		[http(from: header, name: If-None-Match)]
+		ifNotETag: string;
+	}:
+	{
+		/// The requested external widget.
+		[http(from: body)]
+		externalWidget: ExternalWidget;
+
+		/// The ETag of the external widget.
+		[http(from: header)]
+		eTag: string;
+
+		/// The external widget still has the specified ETag.
+		[http(from: body, code: 304)]
+		notModified: boolean;
+	}
+
+	/// Deletes the specified external widget.
+	[http(method: DELETE, path: "/externalWidgets/{id}", code: 204)]
+	[tag(name: externalWidgets)]
+	method deleteExternalWidget
+	{
+		/// The external widget ID.
+		[validate(value: 0..)]
+		id: int32;
+
+		/// Don't delete the external widget unless it has this ETag.
+		[http(from: header, name: If-Match)]
+		ifETag: string;
+	}:
+	{
+		/// The external widget was not found.
+		[http(from: body, code: 404)]
+		notFound: boolean;
+
+		/// The external widget no longer has the specified ETag.
+		[http(from: body, code: 409)]
+		conflict: boolean;
+	}
+
+	/// Gets the specified external widgets.
+	[http(method: POST, path: "/externalWidgets/get")]
+	[tag(name: externalWidgets)]
+	method getExternalWidgetBatch
+	{
+		/// The IDs of the external widgets to return.
+		[http(from: body), required, validate(count: ..10)]
+		ids: int32[];
+	}:
+	{
+		/// The external widget results.
+		[http(from: body)]
+		results: result<ExternalWidget>[];
+	}
```

Comment:
I appreciate the sentiment, but I'm not sure this feature merits this much churn in the conformance API. Ideally one new method would be enough.